File: Leetcode/Strings/First Unique Character in a String/FirstUniqueCharacterInString.py
# Method - 1
from collections import Counter
class Solution:
    def firstUniqChar(self, s: str) -> int:
        charCount = Counter(s)
        for i in s:
            if charCount[i] == 1:
                return s.index(i)
        return -1

# Time Complexity - O(n)
# Space Complexity - O(1)

# Method - 2
class Solution:
    def firstUniqChar(self, s: str) -> int:
        for i in s:
            if s.count(i) == 1:
                return s.index(i)
        return -1

# Time Complexity - O(n**2)
# Space Complexity - O(n)

File: LeetCode/Top Interview Questions/Top_Interview_Questions_Easy/Strings/First Unique Character in a String/FirstUniqueCharacterinString.cpp
#include<iostream>
#include<string.h>
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int firstUniqChar(string s) {
        int size = s.length();
        vector<int> map(26, 0);
        for(int i=0; i<size; i++)
        {
            map[s[i] - 'a']++;
        }
        for(int i=0; i<size; i++)
        {
            if(map[s[i] - 'a'] == 1)
                return i;
        }       
        return -1;
    }
};

// Time Complexity - O(n) -- Since we go through the string
// of length n 2 times
// Space Complexity - O(1) -- Since each English Alphabet contains 26 letters